extensions/resume.md

```diff
@@ -0,0 +1,262 @@
+
+Capability negotiation example:
+
+    C: CAP LS
+    S: CAP * LS :multi-prefix draft/resume-0.3
+    C: CAP REQ :draft/resume-0.3
+    S: CAP * ACK :draft/resume-0.3
+    S: RESUME TOKEN JKbAypzFiovffzuD8VEfcs6bOLrXsSenxsyZNt87Bb0YaQlyOzuQciz6cB2R
+
+### Commands
+
+#### `RESUME` Command
+
+This command MAY ONLY be sent by a client during connection registration, and indicates that the client wishes to resume their session.
+
+    RESUME <token> [timestamp]
+
+`<token>` is the old connection's resume token.
+
+`<timestamp>`, if given, is a timestamp indicating when the client received the last message from the server on the old connection. This timestamp uses the same format as the IRCv3 `server-time` extension (i.e. `YYYY-MM-DDThh:mm:ss.sssZ`, or in UTC using extended format as specified by ISO 8601:2004(E) 4.3.2), and is passed to other clients to indicates how long the disconnection lasted.
+
+### Messages
+
+The `RESUME` message is used to indicate information and the result of a `RESUME` attempt back to the client who attempted it. This message has various sub-messages, indicated below.
+
+If a client does not recognise a `RESUME` subcommand they receive, they `MUST` silently ignore the subcommand.
+
+#### `RESUME SUCCESS` Message
+
+    RESUME SUCCESS <oldnick>
+
+The `SUCCESS` message indicates that the attempt to resume the connection was successful, registration will now complete and the client will get the current state replayed to them.
+
+#### `RESUME WARN` Message
+
+    RESUME WARN <info>
+
+The `WARN` message indicates that there was a warning about the resume attempt, but not that it has succeeded or failed (it will be followed up by a `SUCCESS` or `ERR`). `<info>` is an information text string indicating the reson for the warning (e.g.: `"Chat history may be truncated"`).
+
+#### `RESUME ERR` Message
+
+    RESUME ERR <info>
+
+The `ERR` message indicates that the resume attempt failed. `<info>` is an information text string indicating the reson for the failure (e.g.: `"Cannot resume connection, token is not valid"`).
+
+#### `RESUMED` Message
+
+    :nick!olduser@oldhost RESUMED <user> <host> [timestamp]
+
+This message is sent to indicate that a client has reconnected. `<nick>`, `<olduser>`, and `<oldhost>` indicate the client that has reconnected, and are the details of the old client. `<user>` and `<host>` indicate the reconnecting client's new username and hostname, and the client MUST process this information as they would a regular [`CHGHOST`](https://ircv3.net/specs/extensions/chghost-3.2.html) message. `<timestamp>`, if given, is a timestamp of the form described above which indicates the last message received by the reconnecting clent.
+
+Upon receiving a `RESUMED` message, clients MUST display in some way that the given user has reconnected. If `<timestamp>` is given, clients SHOULD use this to display how much message history seems to have been lost.
```

Comment:
I think this is better left as a decision for clients to make, clients may wish not to display reconnects. They may not be displaying JOINs and QUITs at all

Reply:
Makes sense. I'll make the text here more exact as to why clients **should** make sure their users know that a reconnection happened and history might've been lost, and less demanding as it is right now.

Reply:
This is now done